File: Baekjoon/Class3/02.1620 나는야 포켓몬 마스터 이다솜.py
import sys
input = sys.stdin.readline

class Pokedex:
    def __init__(self):
        self._size = 0
        self._index_to_name = ["Dummy"] #0번 인덱스 더미 데이터
        self._name_to_index = {}
    def add(self, name):
        self._size += 1
        self._index_to_name.append(name)
        self._name_to_index[name] = self._size
    def get_name(self, index):
        return self._index_to_name[index]
    def get_index(self, name):
        return self._name_to_index[name]

def main():
    pokedex = Pokedex()
    n, m = map(int, input().split())
    out = []

    for _ in range(n): #이름 등록
        pokedex.add(input().rstrip())

    for _ in range(m): #검색
        target = input().rstrip()
        if(target[0].isdigit()): #인덱스 검색
            out.append(pokedex.get_name(int(target)))
        else: #이름 검색
            out.append(str(pokedex.get_index(target)))

    print("\n".join(out))

main()

File: Baekjoon/Class3/02.1620 나는야 포켓몬 마스터 이다솜.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

class Pokedex {
    #size
    #indexToName
    #nameToIndex
    constructor(reserveSize) {
        this.#size = 0
        this.#indexToName = new Array(reserveSize + 1)
        this.#indexToName[0] = "Dummy"; // 0번 인덱스 더미 데이터
        this.#nameToIndex = new Map();
    }
    add(name) {
        this.#size++;
        this.#indexToName[this.#size] = name;
        this.#nameToIndex.set(name, this.#size);
    }
    getName(index) {
        return this.#indexToName[index];
    }
    getIndex(name) {
        return this.#nameToIndex.get(name);
    }
}

const [n, m] = input[0].split(" ").map(Number);
const pokedex = new Pokedex(n);
const out = []

for(let i = 1; i <= n; i++) // 이름 등록
    pokedex.add(input[i].trim());

for(let i = n + 1; i <= n + m; i++) { // 검색
    const target = input[i];
    if(isNaN(target[0])) // 이름 검색
        out.push(pokedex.getIndex(target.trim()));
    else // 인덱스 검색
        out.push(pokedex.getName(Number(target)));
}
console.log(out.join("\n"));